filebeat/harvester/reader/encoder.go

```diff
@@ -0,0 +1,35 @@
+package reader
+
+import (
+	"io"
+	"time"
+
+	"github.com/elastic/beats/filebeat/harvester/encoding"
+)
+
+// Encode reader produces lines by reading lines from an io.Reader
+// through a decoder converting the reader it's encoding to utf-8.
+type Encode struct {
```

Comment:
hm... I think we should name the Decode, Conver or Transform. This reader converts input encoding into utf8